File: speckit-chat-vscode/src/model/chat-models.ts
export enum ChatRunStatus {
    RUNNING = 'RUNNING',
    COMPLETED = 'COMPLETED',
    FAILED = 'FAILED',
    CANCELLED = 'CANCELLED',
}

export class ChatRun {
    status: ChatRunStatus = ChatRunStatus.RUNNING;
    durationMs = 0;
    sessionId: string | undefined;
    errorMessage: string | undefined;

    constructor(
        public readonly agent: string,
        public readonly prompt: string,
        public readonly branch: string,
        public readonly startTimeMillis: number = Date.now(),
    ) {}
}

File: speckit-chat-plugin/src/main/kotlin/com/speckit/plugin/model/ChatModels.kt
package com.speckit.plugin.model

enum class ChatRunStatus { RUNNING, COMPLETED, FAILED, CANCELLED }

class ChatRun(
    val agent: String,
    val prompt: String,
    val branch: String,
    val startTimeMillis: Long = System.currentTimeMillis()
) {
    @Volatile var status: ChatRunStatus = ChatRunStatus.RUNNING
    @Volatile var durationMs: Long = 0
    @Volatile var sessionId: String? = null
    @Volatile var errorMessage: String? = null
}
